tutorials/tour/_posts/2017-02-13-variances.md

```diff
@@ -10,31 +10,114 @@ next-page: upper-type-bounds
 previous-page: generic-classes
 ---
 
-Scala supports variance annotations of type parameters of [generic classes](generic-classes.html). In contrast to Java 5 (aka. [JDK 1.5](http://java.sun.com/j2se/1.5/)), variance annotations may be added when a class abstraction is defined, whereas in Java 5, variance annotations are given by clients when a class abstraction is used.
+Scala supports variance annotations of type parameters of [generic classes](generic-classes.html) similar to C#. In contrast to Java 5 (aka. JDK 1.5), variance annotations may be added when a class abstraction is defined, whereas in Java 5, variance annotations are given by clients when a class abstraction is used.
 
-In the page about [generic classes](generic-classes.html) an example for a mutable stack was given. We explained that the type defined by the class `Stack[T]` is subject to invariant subtyping regarding the type parameter. This can restrict the reuse of the class abstraction. We now derive a functional (i.e. immutable) implementation for stacks which does not have this restriction. Please note that this is an advanced example which combines the use of [polymorphic methods](polymorphic-methods.html), [lower type bounds](lower-type-bounds.html), and covariant type parameter annotations in a non-trivial fashion. Furthermore we make use of [inner classes](inner-classes.html) to chain the stack elements without explicit links.
+### Covariance
+
+A type parameter `T` of a generic class can be made covariant by using the annotation `+T`. For some class `Request[T]`, making  `T` covariant implies that for two types `A` and `B` where `A` is a sub-type of `B`, then `Request[A]` is a sub-type of `Request[B]`. This allows us to make very useful and intuitive sub-typing relationships using generics.
+
+Consider this simple class structure:
+
+```tut
+abstract class Animal {
+  def name: String
+}
+case class Cat(name: String) extends Animal
+case class Dog(name: String) extends Animal
+```
+
+Both `Cat` and `Dog` are sub-types of `Animal`. The Scala standard library has a generic immutable `List[T]` class, where the type parameter `T` is covariant. This means that a `List[Cat]` is a `List[Animal]` and a `List[Dog]` is also a `List[Animal]`. Intuitively, it makes sense that a list of cats and a list of dogs are each lists of animals, and you should be able to substitute either of them for a `List[Animal]`.
+
+In the following example, the method `printAnimalNames` will accept a list of animals as an argument and print their names each on a new line. If `List[T]` were not covariant, the last two method calls would not compile.
+
+```tut
+object CovarianceTest extends App {
+  def printAnimalNames(animals: List[Animal]): Unit = {
+    animals.foreach { animal =>
+      println(animal.name)
+    }
+  }
+  
+  val cats: List[Cat] = List(Cat("Whiskers"), Cat("Tom"))
+  val dogs: List[Dog] = List(Dog("Fido"), Dog("Rex"))
+  
+  printAnimalNames(cats)
```

Comment:
Since it prints out multiple lines, what might look acceptable?

Something like this?
```
// prints:
// Whiskers
// Tom
printAnimalNames(cats)
```